### PR TITLE
#167313362 social login 

### PR DESCRIPTION
**Pull request by salviosage**
#### What does this PR do?

- social login 

#### Description of Task to be completed?

- user should log in using the social platform 

- after login user should get his profile and reflected in the navbar

#### How should this be manually tested?

- by cloning this repo and checkout to this branch

- go to the /login route click social login you want to use 

#### What are the relevant pivotal tracker stories?

- [#167313362](https://www.pivotaltracker.com/story/show/167313362)

#### Screenshots (if appropriate)
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/SocialButtons/SocialButton.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Button from '../common/button/Button';
+// import Footer from '../footer/Footer';
+// import NavBar from '../navbar/navbar';
+
+
+const SocialButtons = () => {
+  const { BACKEND_URL } = process.env;
```
Comment (SokoPaulSokool) [inline src/components/SocialButtons/SocialButton.js]: Use the URL in the action types as we agreed

File: src/components/signup/SignupSvg.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export default function SignupSvg() {
+  return (
+    <div className="msg-svg" style={{ commonStyle }}>
+      <div className="msg-svg__message">
+        <p>If you already have an account!</p>
+      </div>
+      <div className="msg-svg__Login-svg">
+        <a href="/login">
+          <svg
```
Comment (SokoPaulSokool) [inline src/components/signup/SignupSvg.js]: Put this SVG in its own assets. Ask you, teammates, how to do it.

File: src/redux/actions/socialAuth/socialAuth.test.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+const profile = {
+  username: 'admin',
+  image: 'fakepath/image.jpg',
+  bio: 'sample bio message',
+};
+describe(' SocialAuth Actions', () => {
+  const error = {
+    message: 'error occured',
+  };
+  // const url = 'http://localhost:3000/api/v1/auth/google/callback';
```
Comment (SokoPaulSokool) [inline src/redux/actions/socialAuth/socialAuth.test.js]: Don't forget to remove commented code

**Review by SokoPaulSokool — CHANGES_REQUESTED** [review]

File: src/components/SocialButtons/SocialButton.js (hunk 1 above)
Comment (salviosage) [reply src/components/SocialButtons/SocialButton.js]: resolved

File: src/redux/actions/socialAuth/socialAuth.test.js (hunk 3 above)
Comment (salviosage) [reply src/redux/actions/socialAuth/socialAuth.test.js]: resolved

File: src/components/SocialButtons/SocialButton.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        </Button>
+      </a>
+    </div>
+
+  ));
+
+  return (
+    <>
+
+      {/* <NavBar isLogedin="true" /> */}
```
Comment (minega25) [inline src/components/SocialButtons/SocialButton.js]: Please remove these unused comments

File: src/components/SocialButtons/SocialButton.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      <div className="text-center social-btn">
+        {socialMediaLogin}
+      </div>
+      {/* <Footer /> */}
```
Comment (minega25) [inline src/components/SocialButtons/SocialButton.js]: Same here.

File: webpack.config.js [hunk 6]
```diff
@@ -9,9 +9,18 @@ const distPath = path.join(__dirname, './dist');
...
   mode: env.environment,
   devServer: {
+    historyApiFallback: {
+      disableDotRule: true,
+    },
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+      https: true,
+      // 'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
+      // 'Access-Control-Allow-Headers': 'X-Requested-With, content-type, Authorization',
```
Comment (minega25) [inline webpack.config.js]: Could you also work on removing these as well.

**Review by minega25 — CHANGES_REQUESTED** [review]
Excellent work. Kindly remove all these unused comments

File: src/components/SocialButtons/SocialButton.js (hunk 5 above)
Comment (salviosage) [reply src/components/SocialButtons/SocialButton.js]: worked on it thanks

File: webpack.config.js (hunk 6 above)
Comment (salviosage) [reply webpack.config.js]: resolved

File: src/components/SocialButtons/SocialButton.js (hunk 4 above)
Comment (salviosage) [reply src/components/SocialButtons/SocialButton.js]: resolved

**Review by habinezadalvan — CHANGES_REQUESTED** [review]
Nice work @salviosage, but I think that it would be better if you follow the mockup. Use the same color and arrange the social buttons as they are on the mockup. Thanks

@salviosage Great Work.

**Review by minega25 — APPROVED** [review]
LGTM.

File: src/components/app.js [hunk 7]
```diff
@@ -4,14 +4,15 @@ import { BrowserRouter } from 'react-router-dom';
...
+import Footer from './common/footer/Footer';
 
 const App = () => (
   <div>
     <BrowserRouter>
       <ToastContainer />
       <Routes />
     </BrowserRouter>
-    ,
+    <Footer />
```
Comment (Anguandia) [inline src/components/app.js:15]: Good work @salviosage, if you will, please confirm if there's a Footer component in common.

**Review by Anguandia — APPROVED** [review]

File: src/components/app.js (hunk 7 above)
Comment (salviosage) [reply src/components/app.js:15]: yeah I created it, @Anguandia

File: src/components/app.js [hunk 8]
```diff
@@ -4,14 +4,15 @@ import { BrowserRouter } from 'react-router-dom';
 import 'bootstrap/dist/css/bootstrap.css';
 import 'react-toastify/dist/ReactToastify.css';
 import Routes from './Routes';
+import Footer from './common/footer/Footer';
 
 const App = () => (
   <div>
```
Comment (nshutijonathan) [inline src/components/app.js:10]: I appreciate how you took the initiative to make a footer component

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by habinezadalvan — APPROVED** [review]
LGTM

**Comment by salviosage** [comment]
> Nice work @salviosage, but I think that it would be better if you follow the mockup. Use the same color and arrange the social buttons as they are on the mockup. Thanks
> 
> @salviosage Great Work.

worked on it, Thanks

**Review by SokoPaulSokool — APPROVED** [review]
Good work

**Pull request merged**